bundles/org.eclipse.search.core/search/org/eclipse/search/internal/core/text/IDirtyFileSearchParticipant.java

```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.search.internal.core.text;
+
+import java.util.Map;
+
+import org.eclipse.core.resources.IFile;
+
+import org.eclipse.jface.text.IDocument;
+
+public interface IDirtyFileSearchParticipant {
+	/**
+	 * Discover a list of dirty IFile resources and the current content of those
+	 * resources in the dirty editor
+	 * 
+	 * @return A map of dirty resources to contents
+	 */
+	Map<IFile, IDocument> findDirtyFiles();
```

Comment:
```suggestion
	Stream<Entry<IFile, IDocument>> dirtyFiles();
```

Reply:
Maybe one should just return dirty documents(!) as fas as I know one can query the file for a document already somewhere (@mickaelistria  ? )

Reply:
Indeed, it seems to be possible via `FileBuffers.getTextFileBufferManager()`, but I wouldn't make it a requirement to use it. This Map is fine at the moment.
Keep in mind that this is an *internal* binding between new UI and core bundles, the goal at this stage is mostly to make it happen without regressions; so I wouldn't worry for internal method signatures.

Reply:
The producer is actually supply open editors (what not necessary must reflect local or remote files), while the consumer is interested in files as an implementation detail so I won't add this aspect to the API.

Reply:
I've accepted the rename from findDirtyFiles() to just dirtyFiles()